crates/alloy-dyn-contract/src/interface.rs

```diff
@@ -0,0 +1,123 @@
+use std::collections::HashMap;
+
+use crate::{
+    error::{Error, Result},
+    instance::ContractInstance,
+};
+use alloy_dyn_abi::{DynSolValue, FunctionExt, JsonAbiExt};
+use alloy_json_abi::{Function, JsonAbi};
+use alloy_primitives::{Address, Selector};
+
+/// A smart contract interface.
+#[derive(Debug, Clone)]
```

Comment:
Could we add a bit more docs on why it's necessary/more ergonomic to have this type vs extending `JsonAbi`?